nexus/db-queries/src/db/datastore/inventory.rs

```diff
@@ -2794,45 +2795,12 @@ mod test {
         logctx.cleanup_successful();
     }
 
-    /// Creates a representative collection, deletes it, and walks through
-    /// tables to ensure that the subcomponents of the inventory have been
-    /// deleted.
-    ///
-    /// NOTE: This test depends on the naming convention "inv_" prefix name
-    /// to identify pieces of the inventory.
-    #[tokio::test]
-    async fn test_inventory_deletion() {
-        // Setup
-        let logctx = dev::test_setup_log("inventory_deletion");
-        let mut db = test_setup_database(&logctx.log).await;
-        let (opctx, datastore) = datastore_test(&logctx, &db).await;
-
-        // Create a representative collection and write it to the database.
-        let Representative { builder, .. } = representative();
-        let collection = builder.build();
-        datastore
-            .inventory_insert_collection(&opctx, &collection)
-            .await
-            .expect("failed to insert collection");
-
-        // Delete that collection we just added
-        datastore
-            .inventory_delete_collection(&opctx, collection.id)
-            .await
-            .expect("failed to prune collections");
-        assert_eq!(
-            datastore
-                .inventory_collections()
-                .await
-                .unwrap()
-                .iter()
-                .map(|c| c.id.into())
-                .collect::<Vec<CollectionUuid>>(),
-            &[]
-        );
-
-        // Read all "inv_" tables and ensure that they're empty
+    enum AllInvTables {
+        AreEmpty,
+        ArePopulated,
+    }
 
+    async fn check_all_inv_tables(datastore: &DataStore, status: AllInvTables) {
```

Comment:
Nit - maybe
```suggestion
    #[track_caller]
    async fn check_all_inv_tables(datastore: &DataStore, status: AllInvTables) {
```

so assertion failures get attached to the tests that call this helper?

Reply:
mmm gave this a shot, turns out it doesn't really like being attached to async functions.

```
warning: `#[track_caller]` on async functions is a no-op
```

I'll do the work to plumb an anyhow::Error out of here, with context.

Reply:
With the new result-based version, I can see this error when I intentionally omit a zpool from the representative collection:

```
--- STDERR:              nexus-db-queries db::datastore::inventory::test::test_inventory_deletion ---
log file: /tmp/nexus_db_queries-391529f8ced893bc-inventory_deletion.3760136.0.log
note: configured to log to "/tmp/nexus_db_queries-391529f8ced893bc-inventory_deletion.3760136.0.log"
thread 'db::datastore::inventory::test::test_inventory_deletion' panicked at nexus/db-queries/src/db/datastore/inventory.rs:2887:14:
All inv_... tables should be populated by representative collection: Found table without entries: inv_zpool

```